clients/cloud/java/src/main/java/io/confluent/examples/clients/ProducerExample.java

```diff
@@ -16,57 +16,86 @@
+          // Ignore if TopicExistsException, which may be valid if topic exists
+          if (!(e.getCause() instanceof TopicExistsException)) {
+              throw new RuntimeException(e);
+          }
+      }
+  }
+
   public static void main(String[] args) throws IOException {
-    if (args.length != 3) {
-      System.out.println("Please provide command line arguments: configPath topic numEvents");
+    if (args.length != 2) {
+      System.out.println("Please provide command line arguments: configPath topic");
       System.exit(-1);
     }
 
-    long events = Long.parseLong(args[2]);
-    String topic = args[1];
-
-    // Load properties from disk.
+    // Load properties from file
     Properties props = loadConfig(args[0]);
 
-    // Add additional properties.
-    props.put("acks", "all");
-    props.put("retries", 200);
-    props.put("batch.size", 16384);
-    props.put("linger.ms", 100);
-    props.put("compression.type", "lz4");
-    props.put("buffer.memory", 33554432);
-    props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
-    props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
-
+    // Create topic if needed
+    String topic = args[1];
+    createTopic(topic, 12, 3, props);
 
-    Producer<String, String> producer = new KafkaProducer<String, String>(props);
+    // Add additional properties.
+    props.put(ProducerConfig.ACKS_CONFIG, "all");
+    props.put(ProducerConfig.RETRIES_CONFIG, 200);
+    props.put(ProducerConfig.BATCH_SIZE_CONFIG, 16384);
+    props.put(ProducerConfig.LINGER_MS_CONFIG, 100);
+    props.put(ProducerConfig.BUFFER_MEMORY_CONFIG, 33554432);
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonSerializer");
 
-    // Produce sample data.
-    Random rnd = new Random();
-    for (long nEvents = 0; nEvents < events; nEvents++) {
-      long runtime = new Date().getTime();
-      String site = "www.example.com";
-      String ip = "192.168.2." + rnd.nextInt(255);
+    Producer<String, JsonNode> producer = new KafkaProducer<String, JsonNode>(props);
 
-      String record = String.format("%s,%s,%s", runtime, site, ip);
+    // Produce sample data
+    final Long numMessages = 10L;
+    for (Long i = 0L; i < numMessages; i++) {
+      String key = "alice";
+      RecordJSON countRecord = new RecordJSON(i);
+      JsonNode value = MAPPER.valueToTree(countRecord);
 
-      producer.send(new ProducerRecord<>(topic, ip, record));
+      producer.send(new ProducerRecord<String, JsonNode>(topic, key, value));
+      System.out.printf("Produced record: %s\t%s%n", key, value);
     }
 
-    System.out.printf("Successfully produced %s messages to %s.\n", events, topic);
+    System.out.printf("10 messages were successfully produced to topic %s%n", topic);
```

Comment:
since the send was asynch we don't know if the messages were successfully produced.
Suggesting using a callback to send() for error reporting, and calling flush() here to wait for all messages to deliver/fail.

Reply:
Implemented Callback